File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/classes/GlideBuilt.java
package com.atomykcoder.atomykplay.classes;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import com.atomykcoder.atomykplay.R;
import com.bumptech.glide.Glide;
import com.bumptech.glide.TransitionOptions;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;

public class GlideBuilt {

    private final Context context;

    public GlideBuilt(Context _context) {
        context = _context;
    }

    public void glide(String uri, int placeholderImage, ImageView imageView, int image_measure) {
        try {
            Glide.with(context).load(uri).apply(new RequestOptions().placeholder(placeholderImage).error(R.drawable.ic_music_thumbnail))
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
                    .override(image_measure, image_measure)
                    .into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void glideBitmap(Bitmap newBitmap, int placeholderImage, ImageView imageView, int image_measure,boolean animate) {
        if (newBitmap != null){
            placeholderImage = -1;
        }
        DrawableTransitionOptions drawableTransitionOptions;
        if (animate){
            drawableTransitionOptions = DrawableTransitionOptions.withCrossFade(300);
        }else {
            drawableTransitionOptions = new DrawableTransitionOptions().dontTransition();
        }
        try {
            Glide.with(context).load(newBitmap).apply(new RequestOptions().placeholder(placeholderImage).error(R.drawable.ic_music_thumbnail))
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
                    .transition(drawableTransitionOptions)
                    .override(image_measure, image_measure)
                    .into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/classes/GlideBuilt.kt
package com.atomykcoder.atomykplay.classes

import android.content.Context
import android.graphics.Bitmap
import android.widget.ImageView
import com.atomykcoder.atomykplay.R
import com.atomykcoder.atomykplay.helperFunctions.AudioFileCover
import com.atomykcoder.atomykplay.helperFunctions.GlideApp
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions

class GlideBuilt(private val context: Context) {
    private val requestOptions = RequestOptions()
    fun loadFromUri(
        uri: String?,
        placeholderImage: Int,
        imageView: ImageView?,
        image_measure: Int
    ) {
        Glide.with(context).load(uri).apply(
            RequestOptions().placeholder(placeholderImage).error(R.drawable.ic_music)
        )
            .diskCacheStrategy(DiskCacheStrategy.NONE)
            .skipMemoryCache(true)
            .transition(DrawableTransitionOptions.withCrossFade(300))
            .override(image_measure, image_measure)
            .into(imageView!!)
    }

    fun loadFromRes(res: Int?, imageView: ImageView?) {
        Glide.with(context).load(res)
            .diskCacheStrategy(DiskCacheStrategy.NONE)
            .skipMemoryCache(true)
            .into(imageView!!)
    }

    fun loadAlbumArt(
        path: String?,
        placeholderImage: Int,
        imageView: ImageView?,
        image_measure: Int,
        animate: Boolean
    ) {

        val drawableTransitionOptions: DrawableTransitionOptions = if (animate) {
            DrawableTransitionOptions.withCrossFade(300)
        } else {
            DrawableTransitionOptions().dontTransition()
        }
        imageView?.let {
            GlideApp.with(context).load(path?.let { AudioFileCover(it) }).apply(
                requestOptions.placeholder(
                    placeholderImage
                ).error(R.drawable.ic_music)
            )
                .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
                .skipMemoryCache(true)
                .transition(drawableTransitionOptions)
                .override(image_measure, image_measure)
                .into(it)
        }
    }

    fun loadFromBitmap(
        bitmap: Bitmap?,
        placeholderImage: Int,
        imageView: ImageView?,
        image_measure: Int,
        animate: Boolean
    ) {

        val shadowed = if (bitmap == null) {
            placeholderImage
        } else {
            0
        }
        val drawableTransitionOptions: DrawableTransitionOptions = if (animate) {
            DrawableTransitionOptions.withCrossFade(300)
        } else {
            DrawableTransitionOptions().dontTransition()
        }
        try {
            imageView?.let {
                Glide.with(context).load(bitmap).apply(
                    requestOptions.placeholder(
                        shadowed
                    ).error(R.drawable.ic_music)
                )
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
                    .skipMemoryCache(true)
                    .transition(drawableTransitionOptions)
                    .override(image_measure, image_measure)
                    .into(it)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun pauseRequest() {
        if (!Glide.with(context).isPaused) {
            Glide.with(context).pauseRequests()
        }
    }

    fun resumeRequest() {
        if (Glide.with(context).isPaused) {
            Glide.with(context).resumeRequests()
        }
    }
}